### ISSUE TITLE
Remove SSL Required option from Realms

### ISSUE THREAD

**Issue by pedroigor** [keycloak/keycloak#17126]
### Description

We probably want to remove the SSL Required option from realm settings and instead decide if HTTP:

* It does not make sense to force TLS for some realms and not for others
* We can infer if TLS is required or not based on the server configuration like proxy, hostname, and http-enabled options.

I'm not 100% sure if we need this property at all as well as checks for SSL required in our endpoints. Keycloak should always be exposed through TLS and if not is because users must know what they are doing(e.g.: some corner case where HTTP is required). It also introduces additional complexity to our endpoints and additional care when working with them as you need to remember to make those checks.

### Discussion

https://github.com/keycloak/keycloak/issues/12572

### Motivation

* Make realm settings simpler
* Avoid additional complexity in our endpoints

### Details

@stianst As suggested by you, in the worst case we can have some kind of filter to enforce TLS, rather than having the logic hard coded into our endpoints.

**Comment by pedroigor** [comment]
@stianst @mposolda FYI.

**Comment by mposolda** [comment]
I agree that realm option `SSL Required` does not makes too much sense now. AFAIK this is legacy option introduced in early Keycloak days, but with all the capabilities of quarkus distribution (and the fact it is SSL by default) makes this option less sense.

But it will be good to likely have SSL checked at some filter/interceptor to make sure it is handled for all the requests sent to Keycloak? Possibly `QuarkusRequestFilter` is the ideal place?

How about backwards compatibility? It seems to me that:
- For the case when `http-enabled` for the server is false, we don't need to care about backwards compatibility and just check https for all the requests (at the filter level)
- For the case when `http-enabled` is true, we may need to care about backwards compatibility as in case that some realms have `ssl required` switched to `on` or `external` , SSL is currently enforced... So if we remove all checks for `SSL required` from the individual endpoints, we still need to check this at the filter level. Do we need another option like `http-allowed-from` at the server level for backwards compatibility? I think it can be set to:
-- nothing in case of `http-enabled` is false
-- something like `127.*,10.*,192.168.*` in case `http-enabled` is true

WDYT?
